TicketsDemo.EF/Repositories/CSVTrainRepository.cs

```diff
@@ -0,0 +1,126 @@
+            using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
+            {
+                csv.Read();
+                csv.ReadHeader();
+                while (csv.Read())
+                {
+                    if (csv.GetField<int>("Id") == id)
+                    {
+                        train = new Train
+                        {
+                            Id = csv.GetField<int>("Id"),
+                            Number = csv.GetField<int>("Number"),
+                            StartLocation = csv.GetField("StartLocation"),
+                            EndLocation = csv.GetField("EndLocation"),
+                            Carriages = new List<Carriage>()
+                        };
+                    }
+                }
+            }
+
+            using (var reader = new StreamReader(_path + "carriages.csv"))
+            using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
+            {
+                csv.Read();
+                csv.ReadHeader();
+                while (csv.Read())
+                {
+                    if (csv.GetField<int>("TrainId") == train.Id)
+                    {
+                        var carriage = new Carriage
+                        {
+                            Id = csv.GetField<int>("Id"),
+                            Type = (CarriageType)csv.GetField<int>("Type"),
+                            DefaultPrice = csv.GetField<int>("DefaultPrice"),
+                            TrainId = csv.GetField<int>("TrainId"),
+                            Number = csv.GetField<int>("Number"),
+                            Places = new List<Place>()
+                        };
+                        train.Carriages.Add(carriage);
+                    }
+                }
+            }
+
+            using (var reader = new StreamReader(_path + "places.csv"))
+            using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
+            {
+                csv.Read();
+                csv.ReadHeader();
+                while (csv.Read())
+                {
+                    if (csv.GetField<int>("CarriageId") == train.Carriages.FirstOrDefault().Id)
+                    {
+                       var place = new Place
+                        {
+                            Id = csv.GetField<int>("Id"),
+                            Number = csv.GetField<int>("Number"),
+                            PriceMultiplier = csv.GetField<decimal>("PriceMultiplier"),
+                            CarriageId = csv.GetField<int>("CarriageId"),
+                            Carriage = train.Carriages.FirstOrDefault(x=>x.Id == csv.GetField<int>("CarriageId"))
+                       };
+                        train.Carriages.FirstOrDefault(x => x.Id == place.CarriageId).Places.Add(place);
+                    }
+                }
+            }
+            return train;
+        }
+
+        public void CreateTrain(Train train)
```

Comment:
there should be no methods throwing "not implemented exception".

Please implement these methods.

I guess that when you start - it will become essential that you need to create some sort of configuration in order not-to-duplicate the string key of csv column multiple times.
Such duplication is very very issue prone.